dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageServiceImpl.java

```diff
@@ -326,15 +328,16 @@ public void cleanup(boolean deleteDbRecords, boolean verbose) throws SQLExceptio
         }
     }
 
-    public Long getLastModified(Bitstream bitstream) {
-        Map wantedMetadata = new HashMap();
-        wantedMetadata.put("modified", null);
-        try {
-            wantedMetadata = stores.get(incoming).about(bitstream, wantedMetadata);
```

Comment:
This was the problematic line...note how it hardcodes incoming. This PR corrects this call to use getStoreNumber() so the request is correctly delegated.